Plugin/Catalog/Block/Toolbar.php

```diff
@@ -188,6 +194,19 @@ public function afterSetCollection(// phpcs:ignore EcgM2.Plugins.Plugin.PluginWa
      */
     private function getCmpResult(FacetInterface $facets, $start, $limit)
     {
+        if ($this->pageSize != -1) {
```

Comment:
It would be better to create a new method to return the limit. It will be confusing to pass a specified limit to `getCmpResult` and then alter it inside the method.

Reply:
Or would it be better to just check for the pageSize in the same function, before these lines (line 148): 
`
                $result = $this->getCmpResult(
                    $this->buildWebFacetService->getFacets(),
                    $this->getCurrentPageNumber()-1,
                    $subjectCollection->getPageSize()
                );
`

Reply:
Create a new one, it will make the code more readable

Reply:
Done